WordPress/gutenberg
# Abstract keyboard shorcuts for heading to paragraph transform and vice-versa

Part of https://github.com/WordPress/gutenberg/issues/60567

## What?

As a preparatory work for https://github.com/WordPress/gutenberg/issues/60567, this pull requests seeks to clean up and avoid code duplication for the keyboard shortcuts that are responsible for transforming heading levels and paragraph to heading and vice-versa.

Currently, the code responsible for this feature is duplicated in 4 different locations in the codebase: post editor, sit editor, widgets editor and customize widgets.

I'd appreciate some feedback about the 'architectural' choice of _where_ to move this code. See the 'How' below.
Cc @youknowriad @WordPress/gutenberg-core  when you have a chance 🙏🏽 

Additionally, this PR adds a new keyboard shortcut alias `7` to transform a Heading to Paragraph, for consistency with the Classic Editor where this shortcut has always used the number 7.

## Why?

Code duplication brings in maintenance cost and potential bugs and inconsistencies. Havin a centralized, unique, code responsible for a feature allow to easily change it, extend it, improve it, test it.

## How?

- In a first moment I thought to abstract the related code to a new `BlockKeyboardShortcuts` component and add that to the stack. This was maybe cleaner but it added a new component.
- Then I discovered the other keyboard shortcuts for actions related to blocks (duplicate, move, etc.) are all already registered in the block-editor `KeyboardShortcuts` component and their 'callbacks' use a custom implementation in the `BlockTools` component. In the last commit I moved everything to follow this pattern. However, I'm not sure this existing implementation is ideal, as it mixes a component that does many other things (maybe too many) with the keyboard shortcuts.

Any suggestion on which of the two approaches would be better is welcome.

## Testing Instructions




- Run `npm install` in your terminal before building, as the 'block-library' package has now a new dependency.
- Go to the Post Editor and select a Heading block.
- Check the following keyboard shortcuts work as expected (assuming macOS):
  - Control + Option + 1-6 change the heading level ot transform a Paragraph to a Heading with that level.
  - Control + Option + 0  and Control + Option + 7 change a Heading to a Paragraph.
- Go to the Site Editor and select a Heading block.
- Repeat the test.
- Set Twenty Twenty-One as active theme so that thre Widgets editor adn the Customizer are available.
- Go to Appearance > Widgets.
- Add a heading block to a widgets area and save.
- Select the heading block and repeat the test.
- Go to Customize > Widgets.
- Select the heading block and repeat the test.
- In all 4 cases, open the Keyboard shortcuts modal dialog by pressing Control + Option + H
- Observe the keyboard shortcut `Convert the current heading to a paragraph.` now displays also the new alias with thte key `7`.

### Testing Instructions for Keyboard


## Screenshots or screencast
